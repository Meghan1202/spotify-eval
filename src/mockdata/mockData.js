const mockLike = { like: false, count: 1 };

const mockSongs = [{
  id: '1',
  name: 'song_name',
  genre: { id: '1', name: 'genre_name' },
  artist: { id: '1', name: 'abc' },
  albumArtUrl: 'abc.com',
  publishedAt: '',
  likes: 1,
  isLikedAlready: true,
}];

module.exports = { mockLike, mockSongs };
