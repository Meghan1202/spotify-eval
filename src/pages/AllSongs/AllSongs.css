.Records-Container {
    justify-content: center;
}

.Page-Header {
    display: flex;
    justify-content: space-between;
    margin: 1rem 4rem;
}