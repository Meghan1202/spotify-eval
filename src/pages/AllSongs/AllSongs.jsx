import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import SongCard from '../../components/SongCard/SongCard';
import './AllSongs.css';

const AllSongs = ({
  songs, toggleView, patchLike, updateSongs,
}) => (
  <div className="All-Song-Container">
    <div className="Page-Header">
      <h1>all songs</h1>
      <Link
        to="/genre"
        onClick={() => {
          toggleView();
        }}
      >
        <img src="./assets/icon-genre.svg" alt="Toggle View" />
      </Link>
    </div>
    <div className="Records-Container">
      {
        songs.map((song, index) => (
          <SongCard
            theme={index % 2}
            key={song.id}
            name={song.name}
            albumArt={song.albumArtUrl}
            artist={song.artist.name}
            currentState={song.isLikedAlready}
            id={song.id}
            likes={song.likes}
            patchLike={patchLike}
            updateSongs={updateSongs}
          />
        ))
      }
    </div>
  </div>
);

AllSongs.propTypes = {
  songs: PropTypes.arrayOf(Object).isRequired,
  toggleView: PropTypes.func.isRequired,
  patchLike: PropTypes.func.isRequired,
  updateSongs: PropTypes.func.isRequired,
};

export default AllSongs;
