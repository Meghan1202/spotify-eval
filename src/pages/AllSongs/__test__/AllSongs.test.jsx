import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import AllSongs from '../AllSongs';
import { mockSongs } from '../../../mockdata/mockData';

describe(AllSongs.name, () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <AllSongs
          songs={mockSongs}
          toggleView={jest.fn()}
          patchLike={jest.fn()}
          updateSongs={jest.fn()}
        />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
