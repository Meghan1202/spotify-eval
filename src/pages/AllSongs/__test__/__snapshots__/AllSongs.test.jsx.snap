// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllSongs should match the snapshot 1`] = `
<div>
  <div
    class="All-Song-Container"
  >
    <div
      class="Page-Header"
    >
      <h1>
        all songs
      </h1>
      <a
        href="/genre"
      >
        <img
          alt="Toggle View"
          src="./assets/icon-genre.svg"
        />
      </a>
    </div>
    <div
      class="Records-Container"
    >
      <div
        class="Card-Container-0"
      >
        <img
          alt="poster"
          class="Image-Container"
          src="abc.com"
        />
        <div
          class="Song-Name"
        >
          song_name
        </div>
        <div
          class="Card-Footer"
        >
          <span
            class="Artist-Name"
          >
            abc
          </span>
          <div
            class="Like-Button-Container"
          >
            <input
              alt="Like"
              class="Heart-Button"
              src="./assets/red-heart.svg"
              type="image"
            />
            <div
              class="centered"
            >
              1
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
